### PR TITLE
Bug #13046: using fully iCal4J API for Date and DateTime decoding into calendar import/synchronization services

### PR DESCRIPTION
**Pull request by SilverYoCha**

### CODE REVIEW

File: core-library/src/main/java/org/silverpeas/core/calendar/ical4j/ICal4JRecurrenceCodec.java [hunk 1]
```diff
@@ -188,43 +182,28 @@ public Recurrence decode(final VEvent vEvent, final ZoneId defaultZoneId) {
...
     /* TODO activating following commented forEach after fix of EXDATE UTC management (iCal4J)
        cf. https://github.com/ical4j/ical4j/issues/113 for example
 
        <code>exDates.forEach(exDate -> recurrence
-          .excludeEventOccurrencesStartingAt(iCal4JDateCodec.decode(exDate, defaultZoneId)));</code>
+          .excludeEventOccurrencesStartingAt(iCal4JDateCodec.decode(exDate)));</code>
 
       Deleting following forEach after fix of EXDATE UTC management (iCal4J)
       cf. https://github.com/ical4j/ical4j/issues/113 for example
       */
```
Comment (mmoqui) [inline core-library/src/main/java/org/silverpeas/core/calendar/ical4j/ICal4JRecurrenceCodec.java:204]: Did you check the bug is always here?

Comment (SilverYoCha) [reply core-library/src/main/java/org/silverpeas/core/calendar/ical4j/ICal4JRecurrenceCodec.java:204]: The issue is still open.

File: core-library/src/main/java/org/silverpeas/core/calendar/ical4j/ICal4JImporter.java [hunk 2]
```diff
@@ -107,32 +105,27 @@ public void imports(final ImportDescriptor descriptor,
...
-            vEvents.add(0, vEvent);
-          }
+          ofNullable(vEvent.getRecurrenceId()).ifPresentOrElse(
+              r -> vEvents.add(vEvent),
+              () -> vEvents.add(0, vEvent));
         } else if (component instanceof VTimeZone) {
-          VTimeZone vTimeZone = (VTimeZone) component;
-          zoneId.set(toZoneId(vTimeZone.getTimeZoneId().getValue()));
+          SilverLogger.getLogger(this)
+              .debug("iCalendar VTimeZone ''{0}'' detected", component.getName());
```
[the remark this answers is not in the document]
Comment (SilverYoCha) [reply core-library/src/main/java/org/silverpeas/core/calendar/ical4j/ICal4JImporter.java:128]: I don't think that it is really really time consuming as this code is almost never called (generally some on TimeZones).
But, I will rearrange the code anyway.

File: core-library/src/main/java/org/silverpeas/core/calendar/ical4j/ICal4JRecurrenceCodec.java (hunk 1 above)
Comment (mmoqui) [reply core-library/src/main/java/org/silverpeas/core/calendar/ical4j/ICal4JRecurrenceCodec.java:204]: Yes on Github. But that doesn't mean it isn't fixed (by another way)

Comment (SilverYoCha) [reply core-library/src/main/java/org/silverpeas/core/calendar/ical4j/ICal4JRecurrenceCodec.java]: I will spend time on this.
As the Silverpeas's code can not be removed in any case in order to handle exlusion dates, and also as the code become simplier, there is no more reason finally to keep this comment.

Comment (SilverYoCha) [reply core-library/src/main/java/org/silverpeas/core/calendar/ical4j/ICal4JRecurrenceCodec.java]: Removed.

**Review by mmoqui — APPROVED** [review]

**Pull request merged**